dbt_platform_helper/commands/environment.py

```diff
@@ -594,50 +562,57 @@ def add_maintenance_page(
     env: str,
     services: List[Service],
     allowed_ips: tuple,
-    ip_filter: bool,
     template: str = "default",
 ):
     lb_client = session.client("elbv2")
     maintenance_page_content = get_maintenance_page_template(template)
+    bypass_value = "".join(random.choices(string.ascii_lowercase + string.digits, k=12))
+
+    service_number = 1
 
     for svc in services:
-        target_group_arn = find_target_group(app, env, svc.name)
+        target_group_arn = find_target_group(app, env, svc.name, session)
 
         # not all of an application's services are guaranteed to have been deployed to an environment
         if not target_group_arn:
             continue
 
-        if not ip_filter:
-            user_ip = get_public_ip()
-            allowed_ips = list(allowed_ips) + [user_ip]
-
+        allowed_ips = list(allowed_ips)
+        max_allowed_ips = 100
+        for ip_index, ip in enumerate(allowed_ips):
+            forwarded_rule_priority = service_number * (ip_index + max_allowed_ips)
```

Comment:
For the reasons mentioned on https://github.com/uktrade/platform-tools/pull/473#discussion_r1686196863, this maths should still be 
```suggestion
            forwarded_rule_priority = (service_number * max_allowed_ips) + ip_index
```